vizro-core/tests/unit/vizro/models/_components/form/test_range_slider.py

```diff
@@ -229,16 +229,20 @@ def test_validate_step_invalid(self):
         "marks, expected",
         [
             ({i: str(i) for i in range(0, 10, 5)}, {i: str(i) for i in range(0, 10, 5)}),
-            ({15: 15, 25: 25}, {15.0: "15", 25.0: "25"}),
-            ({"15": 15, "25": 25}, {15.0: "15", 25.0: "25"}),
+            ({15: 15, 25: 25}, {15: "15", 25: "25"}),  # all int
+            ({15.5: 15.5, 25.5: 25.5}, {15.5: "15.5", 25.5: "25.5"}),  # all floats
+            ({15.0: 15, 25.0: 25}, {15: "15", 25: "25"}),  # all floats, but convertible to int
```

Comment:
A mixed case might be good here too
```suggestion
            ({15.0: 15, 25.5: 25}, {15: "15", 25.5: "25"}),
```